### ISSUE TITLE
Clojure compatibility: calling vars

### ISSUE THREAD

**Issue by ikappaki** [basilisp-lang/basilisp#767]
In Clojure, it is possible to call `var`s (useful for calling private fns from another namespace), but basilisp throws an exception if so:

```TypeError: 'Var' object is not callable```

To reproduce
1. Open up the REPL, and  call the the `basilisp.core/list` fn, it throws an exception
```clojure
basilisp.user=> (#'list)
Traceback (most recent call last):
  File "src/basilisp/src/basilisp/cli.py", line 356, in repl
    result = eval_str(lsrc, ctx, ns, eof)
  File "src/basilisp/src/basilisp/cli.py", line 52, in eval_str
    last = compiler.compile_and_exec_form(form, ctx, ns)
  File "basilisp/src/basilisp/lang/compiler/__init__.py", line 187, in compile_and_exec_form
    return getattr(ns.module, final_wrapped_name)()
  File "<REPL Input>", line 1, in __lisp_expr___73
TypeError: 'Var' object is not callable
```
The same works fine in Clojure
``` clojure
user> (#'list)
()
```

I don't think this is an official documented behavior in Clojure, but I have seen it used in the wild, most helpful when one wants to test private fn definitions in tests. There are references to this practice on the internet, such as in 
https://blog.danieljanus.pl/2012/04/25/how-to-call-a-private-function-in-clojure/

The alternative in `basilisp` to achieve this would be to deref the var, which also works in clojure as such
```
basilisp.user=> (@#'list)
()
```

Should basilisp maintain compatibility with Clojure in terms of callable vars, or should the basilisp compatibility documentation be updated to highlight the alternative @ syntax for achieving the same result? Happy to provide a PR for either approach.

Thanks

**Comment by chrisrink10** [comment]
I was unaware of this behavior so did not know to implement it. I think it makes sense to make Vars callable so Basilisp works like Clojure. Thanks!

**Comment by chrisrink10** [comment]
Fixed in #768
